### PR TITLE
Adds interactive mode for `get` command

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -3,21 +3,24 @@ module github.com/tektoncd/hub/api
...
 	github.com/ikawaha/goahttpcheck v1.3.1
 	github.com/joho/godotenv v1.3.0
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/sergi/go-diff v1.2.0 // indirect
-	github.com/spf13/cobra v1.1.1
+	github.com/spf13/cobra v1.1.3
 	github.com/spf13/viper v1.7.0
 	github.com/stretchr/testify v1.6.1
-	github.com/tektoncd/pipeline v0.20.1-0.20210204110343-8c5a751b53ea
+	github.com/tektoncd/pipeline v0.23.0
```
Comment (sm43) [inline api/go.mod:23]: bump to 0.24.1
in sync with tkn

File: api/pkg/cli/cmd/get/get.go [hunk 2]
```diff
@@ -97,10 +99,16 @@ func (opts *options) run() error {
 		return err
 	}
 
-	hubClient := opts.cli.Hub()
+	opts.hubClient = opts.cli.Hub()
+	var err error
 
-	resource := hubClient.GetResource(hub.ResourceOption{
-		Name:    opts.name(),
+	name, err := opts.GetResourceInfo()
```
Comment (sm43) [inline api/pkg/cli/cmd/get/get.go:105]: can we keep similar steps as `info` command?
it was nice instead of keeping inside `opts.GetResourceInfo()`

Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/get/get.go:105]: Yes will update the pr once info gets in

File: api/pkg/cli/cmd/get/task.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func taskCommand(opts *options) *cobra.Command {
 		Annotations: map[string]string{
 			"commandType": "main",
 		},
-		Args: cobra.ExactArgs(1),
+		// Args: cobra.ExactArgs(1),
```
Comment (sm43) [inline api/pkg/cli/cmd/get/task.go:43]: nit

File: api/pkg/cli/cmd/get/get_test.go [hunk 4]
```diff
@@ -180,13 +310,13 @@ func TestGet_WithVersion(t *testing.T) {
...
 		from:    "fruit",
-		version: "0.3",
+		version: "0.2",
 	}
 
 	err := opt.run()
 	assert.NoError(t, err)
 	golden.Assert(t, buf.String(), fmt.Sprintf("%s.golden", t.Name()))
-	assert.Equal(t, gock.IsDone(), true)
+	assert.Equal(t, gock.IsDone(), false)
```
Comment (sm43) [inline api/pkg/cli/cmd/get/get_test.go:319]: why this is changed to false?

Comment (PuneetPunamiya) [reply api/pkg/cli/cmd/get/get_test.go]: I have fixed the test, ptal

**Pull request merged**